core/jobs/batch_jobs/firebase_sync_jobs.py

```
@@ -0,0 +1,60 @@
+    This job runs in two phases:
+    1. Delete ALL records that exist on the Firebase server.
+    2. Import NEW records from the identifiers in Oppia's user & auth models.
+    """
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        if (
+            feature_flag_domain.get_server_mode()
+            == feature_flag_domain.ServerMode.PROD
+        ):
+            job_name = self.__class__.__name__
+            raise PermissionError(f'{job_name} must never be run in production')
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)